edx_solutions_api_integration/users/tests.py

```diff
@@ -2353,3 +2354,203 @@ def test_user_courses_grades_list_get_after_enrollment(self):  # pylint: disable
         self.assertEqual(response.data[0]['current_grade'], 0, 0)
         self.assertEqual(response.data[0]['proforma_grade'], 0, 0)
         self.assertEqual(response.data[0]['complete_status'], False)
+
+
+@override_settings(MODULESTORE=MODULESTORE_CONFIG)
+class UsersProgressApiTests(
+    SignalDisconnectTestMixin, SharedModuleStoreTestCase, APIClientMixin, CourseGradingMixin
+):
+    """ Test suite for User Progress API views """
+
+    @classmethod
+    def setUpClass(cls):
+        super(UsersProgressApiTests, cls).setUpClass()
+        cls.base_courses_uri = '/api/server/courses'
+        cls.base_users_uri = '/api/server/users'
+
+        cls.course_start_date = timezone.now() + relativedelta(days=-1)
+        cls.course_end_date = timezone.now() + relativedelta(days=60)
+        cls.course = CourseFactory.create(
+            start=cls.course_start_date,
+            end=cls.course_end_date,
+        )
+        cls.test_data = '<html>{}</html>'.format(str(uuid.uuid4()))
+
+        cls.chapter = ItemFactory.create(
+            category="chapter",
+            parent_location=cls.course.location,
+            data=cls.test_data,
+            due=cls.course_end_date,
+            display_name="Overview",
+        )
+
+        cls.course_project = ItemFactory.create(
+            category="chapter",
+            parent_location=cls.course.location,
+            data=cls.test_data,
+            display_name="Group Project"
+        )
+
+        cls.course_project2 = ItemFactory.create(
+            category="chapter",
+            parent_location=cls.course.location,
+            data=cls.test_data,
+            display_name="Group Project2"
+        )
+
+        cls.course_content2 = ItemFactory.create(
+            category="sequential",
+            parent_location=cls.chapter.location,
+            data=cls.test_data,
+            display_name="Sequential",
+        )
+
+        cls.content_child2 = ItemFactory.create(
+            category="vertical",
+            parent_location=cls.course_content2.location,
+            data=cls.test_data,
+            display_name="Vertical Sequence"
+        )
+
+        cls.course_content = ItemFactory.create(
+            category="videosequence",
+            parent_location=cls.content_child2.location,
+            data=cls.test_data,
+            display_name="Video_Sequence",
+        )
+
+        cls.content_child = ItemFactory.create(
+            category="video",
+            parent_location=cls.course_content.location,
+            data=cls.test_data,
+            display_name="Video"
+        )
+
+        cls.content_subchild = ItemFactory.create(
+            category="video",
+            parent_location=cls.content_child2.location,
+            data=cls.test_data,
+            display_name="Child Video",
+        )
+
+        cls.overview = ItemFactory.create(
```

Comment:
We propably shoud have simple course structure to test user's progress use cases. No need to have about and other sections. Just a chapter, sequential, veritcal and couple of vidoes should be enough.